app/models/error/ApiError.scala

```diff
@@ -40,16 +41,19 @@ object ApiError {
 
   def convertToErrorCode(apiError: ApiError): Int =
     apiError match {
-      case NotFoundError           => Status.NOT_FOUND
-      case BadRequestError         => Status.BAD_REQUEST
-      case ServiceUnavailableError => Status.SERVICE_UNAVAILABLE
-      case _                       => Status.INTERNAL_SERVER_ERROR
+      case NotFoundError            => Status.NOT_FOUND
+      case UnprocessableEntityError => Status.UNPROCESSABLE_ENTITY
```

Comment:
That is not correct. We don't want to propagate 422 error all the way through to end user. What it was here before is correct, where it catches such errors and returns InternalServerError to end user.

Reply:
So I don't need to give that specific error to user, just return ISE if it picks up a generic error